ffs/scheduler/scheduler_storage.go

```diff
@@ -384,6 +388,12 @@ func (s *Scheduler) executeColdStorage(ctx context.Context, curr ffs.CidInfo, cf
 	// whatever extra deals we need to make that true.
 
 	// Do we need to do some work?
+	if s.sr2RepFactor != nil {
+		cfg.Filecoin.RepFactor, err = s.sr2RepFactor()
+		if err != nil {
+			return ffs.ColdInfo{}, nil, fmt.Errorf("getting SR2 replication factor: %s", err)
+		}
+	}
```

Comment:
OK, a tricky thing about UX.

Without this change, the scheduler would always consider the replication factor in the StorageConfig to know what's the intentional replication factor. That isn't true if the SR2 miner selector is used.

What's more, in SR2 this value is dynamic since lives in a remote file. So the scheduler gets initialized with a `sr2RepFactor` function, that returns the current replication factor of the file, and overrides the StorageConfig with that value. If the SR2 miner selector isn't used then this function would be nil and will work as usual.

This logic will live until we take out this custom miner selector... so would be just deleting this code.

Reply:
Not totally sure of the details, but isn't `RepFactor` normally (not in SR2) used as _input_ to the miner selector so the selector knows how many miners to select? In the case of SR2, the miner selector ignores the `RepFactor` input. Is this change just to force `RepFactor` to be consistent with what the miner selector will do anyways? That is a good thing to do for the sake of consistency and UX etc, but maybe I'm missing something else here?

Reply:
The replication factor is also in the remote JSON file. What this does is doing an alignment of the `RepFactor` value in the `StorageConfig` such that the rest of the scheduler logic takes the correct decisions.